### PR TITLE
monitorcontrol: 4.1.0 -> 4.2.0

### PR DESCRIPTION
**Pull request by cottand**
## Description of changes
See changelog [here](https://github.com/MonitorControl/MonitorControl/releases/tag/v4.2.0)

Specifically:
- DDC support for high-end M2 (Pro/Max/Ultra) systems
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by cbleslie — APPROVED** [review]
Works. Thanks for the update @Cottand 

Also. The version number is 420. 🔥 Blaze it.

File: pkgs/applications/misc/monitorcontrol/default.nix [hunk 1]
```diff
@@ -1,19 +1,40 @@
...
 
-  nativeBuildInputs = [ undmg ];
+  unpackCmd = ''
+    echo "File to unpack: $curSrc"
+    if ! [[ "$curSrc" =~ \.dmg$ ]]; then return 1; fi
+    mnt=$(mktemp -d -t ci-XXXXXXXXXX)
+
+    function finish {
+      echo "Detaching $mnt"
+      /usr/bin/hdiutil detach $mnt -force
```
Comment (wegank) [inline pkgs/applications/misc/monitorcontrol/default.nix:23]: This looks like a crime... Does `7zz x $src` with 7zz from `_7zz` work?

File: pkgs/applications/misc/monitorcontrol/default.nix [hunk 2]
```diff
@@ -1,19 +1,40 @@
...
+  version = "4.2.0";
 
   src = fetchurl {
     url =
       "https://github.com/MonitorControl/${pname}/releases/download/v${version}/MonitorControl.${version}.dmg";
-    sha256 = "iaxM9j78Sq1EH5TCY240N+D5bG6quk2dZj8T7nt9ATo=";
+    sha256 = "Q96uK6wVe1D2uLvWL+pFR6LcmrU7cgmr2Y5tPvvTDgI=";
   };
 
-  nativeBuildInputs = [ undmg ];
```
Comment (cbleslie) [inline pkgs/applications/misc/monitorcontrol/default.nix:16]: For sanity's sake, might good to explain why we're not using `undmg` to unwrap the .app.

**Review by cbleslie — APPROVED** [review]

File: pkgs/applications/misc/monitorcontrol/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/applications/misc/monitorcontrol/default.nix]: 7zz works.

**Review by cbleslie — APPROVED** [review]
@wegank 's changes worked. 👍

**Review by cottand — COMMENTED** [review]
Hey! Sorry for the late reply. I cannot approve a PR I started, but I do prefer @wegank 's solution.

**Comment by cbleslie** [comment]
🎉👏😎

**Pull request merged**